sdk/ml/azure-ai-ml/azure/ai/ml/entities/_compute/_custom_applications.py

```diff
@@ -0,0 +1,226 @@
+# ---------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# ---------------------------------------------------------
+# pylint: disable=protected-access,redefined-builtin
+
+from typing import Dict, List, Optional
+from azure.ai.ml._restclient.v2022_10_01_preview.models import (
+    CustomService,
+    Docker,
+    Endpoint as RestEndpoint,
+    EnvironmentVariable as RestEnvironmentVariable,
+    EnvironmentVariableType as RestEnvironmentVariableType,
+    Image as RestImage,
+    ImageType as RestImageType,
+    Protocol,
+    VolumeDefinition as RestVolumeDefinition,
+    VolumeDefinitionType as RestVolumeDefinitionType,
+)
+from azure.ai.ml.exceptions import ErrorCategory, ErrorTarget, ValidationException
+from azure.ai.ml.constants._compute import CustomApplicationDefaults
+
+
+class ImageSettings:
+    """Specifies an image configuration for a Custom Application.
+
+    :param reference: Image reference URL.
+    :type reference: str
+    """
+
+    def __init__(self, *, reference: str):
+        self.reference = reference
+
+    def _to_rest_object(self) -> RestImage:
+        return RestImage(type=RestImageType.DOCKER, reference=self.reference)
+
+    @classmethod
+    def _from_rest_object(cls, obj: RestImage) -> "ImageSettings":
+        return ImageSettings(reference=obj.reference)
+
+
+class EndpointsSettings:
+    """Specifies an endpoint configuration for a Custom Application.
+
+    :param target: Application port inside the container.
+    :type target: int
+    :param published: Port over which the application is exposed from container.
+    :type published: int
+    """
+
+    def __init__(self, *, target: int, published: int):
+        self.target = target
+        self.published = published
+        self._validate_endpoint_settings()
+
+    def _validate_endpoint_settings(self):
+        invalid_value_error_message = "Value of {} property should be between {} and {}"
+        ports = {
+            CustomApplicationDefaults.TARGET_PORT: self.target,
+            CustomApplicationDefaults.PUBLISHED_PORT: self.published,
+        }
+        min_value = CustomApplicationDefaults.PORT_MIN_VALUE
+        max_value = CustomApplicationDefaults.PORT_MAX_VALUE
+
+        for port_name, port in ports.items():
+            msg = invalid_value_error_message.format(port_name, min_value, max_value)
+            if port < min_value or port > max_value:
```

Comment:
for readability maybe try:
```suggestion
            if not min_value < port < max_value:
```

Reply:
Agreed.